### PR TITLE
PT-3834: Bug and style fixes for pedigree editor

### PR DESCRIPTION
**Pull request by buske**
Most of these were identified via eslint. Some are quite serious, e.g., undefined variables.

### CODE REVIEW

File: components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js [hunk 1]
```diff
@@ -248,7 +250,7 @@ define([
               values.push({'value': v, 'text' : ("0" + v).slice(-2)});
           }
       } else {
-          for (var v = end; v <= start; --v) {
+          for (var v = start; v >= end; --v) {
```
Comment (buske) [inline components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js]: If the loop should really start at `end`, then it needs to be: `for (var v = end; v <= start; ++v) {`

File: components/pedigree/resources/src/main/resources/pedigree/model/import.js [hunk 2]
```diff
@@ -1561,7 +1560,7 @@ define([
                                    properties["comments"] = "(untested)\n" + properties["comments"];
                                }
                                break;
-                           case "O":
+                           case "C":
```
Comment (buske) [inline components/pedigree/resources/src/main/resources/pedigree/model/import.js]: "O" is already a case in this switch statement, and "C" seems like the best fit based on the documentation.

File: components/pedigree/resources/src/main/resources/pedigree/model/positionedGraph.js [hunk 3]
```diff
@@ -13,6 +14,7 @@ define([
         "pedigree/view/ageCalc"
     ], function(
         PedigreeDate,
+        DynamicPositionedGraph,
```
Comment (allasm) [inline components/pedigree/resources/src/main/resources/pedigree/model/positionedGraph.js]: `DynamicPositionedGraph` is only used for (old) testing/.debug code, the correct change is to move the code that uses it

File: components/pedigree/resources/src/main/resources/pedigree/model/positionedGraph.js [hunk 4]
```diff
@@ -3156,38 +3156,17 @@ define([
 
         //=[position]========================================================================
 
-        displayGraph: function(xcoord, message)
```
Comment (allasm) [inline components/pedigree/resources/src/main/resources/pedigree/model/positionedGraph.js]: this is used when debugging layout algorithms (I have som etest code in another epository). I would keep it

File: components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js (hunk 1 above)
Comment (allasm) [reply components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js]: change is correct

File: components/pedigree/resources/src/main/resources/pedigree/model/import.js (hunk 2 above)
Comment (allasm) [reply components/pedigree/resources/src/main/resources/pedigree/model/import.js]: looks correct

**Comment by buske** [comment]
@allasm You suggested moving your test code somewhere else, but I'm not sure where you'd like it. I went ahead and removed it, so the dependency it had could also be removed. You're welcome to add back the test code wherever you want it.

**Comment by allasm** [comment]
Sorry, I was actually going to do all of that myself, did not expect you to go back to this PR. I will do it after Christmas break

**Pull request closed without merging**